crux-mir/src/Mir/Overrides.hs

```diff
@@ -376,17 +411,38 @@ bindFn ::
   Text ->
   CFG MIR blocks args ret ->
   OverrideSim (p sym) sym MIR rtp a r ()
-bindFn symOnline _cs name cfg
+bindFn symOnline cs name cfg
   | (normDefId "crucible::array::symbolic" <> "::_inst") `Text.isPrefixOf` name
   , Empty :> MirSliceRepr (BVRepr w) <- cfgArgTypes cfg
   , UsizeArrayRepr btpr <- cfgReturnType cfg
   , Just Refl <- testEquality w (knownNat @8)
   = bindFnHandle (cfgHandle cfg) $ UseOverride $
     mkOverride' "array::symbolic" (UsizeArrayRepr btpr) (array_symbolic btpr)
+
   | (normDefId "crucible::concretize" <> "::_inst") `Text.isPrefixOf` name
   , Empty :> tpr <- cfgArgTypes cfg
   , Just Refl <- testEquality tpr (cfgReturnType cfg)
   = bindFnHandle (cfgHandle cfg) $ UseOverride $ mkOverride' "concretize" tpr $ concretize symOnline
+
+  | (normDefId "crucible::override_" <> "::_inst") `Text.isPrefixOf` name
+  , Empty :> UnitRepr :> UnitRepr <- cfgArgTypes cfg
+  , UnitRepr <- cfgReturnType cfg
+  = bindFnHandle (cfgHandle cfg) $ UseOverride $
+    mkOverride' "crucible_override_" UnitRepr $ overrideRust cs name
+
+  | (normDefId "crucible::dump_what4" <> "::_inst") `Text.isPrefixOf` name
+  , Empty :> MirSliceRepr (BVRepr w) :> (asBaseType -> AsBaseType btpr) <- cfgArgTypes cfg
+  , Just Refl <- testEquality w (knownNat @8)
+  , UnitRepr <- cfgReturnType cfg
+  = bindFnHandle (cfgHandle cfg) $ UseOverride $
+    mkOverride' "crucible_override_" UnitRepr $ do
+        RegMap (Empty :> RegEntry _ strRef :> RegEntry _ expr) <- getOverrideArgs
+        str <- getString strRef >>= \x -> case x of
+            Just x -> return x
+            Nothing -> fail $ "dump_what4: desc string must be concrete"
+        traceM $ Text.unpack str ++ " = " ++ show (printSymExpr expr)
```

Comment:
Ah, this is the use of `Debug.Trace`. I believe `OverrideSim` wraps `IO`, so you can probably use something like `liftIO $ putStrLn ...` here.